src/main.cpp

```diff
@@ -59,14 +59,54 @@ void install_load_hook(uint32_t* target) {
     // Call orig first
     LOG_DEBUG("il2cpp_init called with: {}", domain_name);
     reinterpret_cast<void (*)(char const*)>(trampoline.address.data())(domain_name);
-    modloader::load_mods();
+    modloader::load_early_mods();
   };
   target_hook.WriteCallback(reinterpret_cast<uint32_t*>(+init_hook));
   target_hook.Finish();
   FLAMINGO_DEBUG("Hook installed! Target: {} (il2cpp_init) now will call: {} (hook), with trampoline: {}",
                  fmt::ptr(target), fmt::ptr(+init_hook), fmt::ptr(trampoline.address.data()));
 }
 
+void install_unity_hook(uint32_t* target) {
+  // TODO: create the hook for hooking libunity.so::PlayerLoadFirstScene
+  // Size of the allocation size for the trampoline in bytes
+  constexpr static auto trampolineSize = 64;
+  // Size of the function we are hooking in instructions
+  constexpr static auto hookSize = 8;
+  // Size of the allocation page to mark as +rwx
+  constexpr static auto kPageSize = 4096ULL;
+  // Mostly throw-away reference
+  size_t trampoline_size = trampolineSize;
+  static auto trampoline = flamingo::TrampolineAllocator::Allocate(trampolineSize);
+  // We write fixups for the first 4 instructions in the target
+  trampoline.WriteHookFixups(target);
+  // Then write the jumpback at instruction 5 to continue the code
+  trampoline.WriteCallback(&target[4]);
+  trampoline.Finish();
+  // Ensure target is writable
+  auto* page_aligned_target = reinterpret_cast<uint32_t*>(reinterpret_cast<uint64_t>(target) & ~(kPageSize - 1));
+  FLAMINGO_DEBUG("Marking target: {} as writable, page aligned: {}", fmt::ptr(target), fmt::ptr(page_aligned_target));
+  if (::mprotect(page_aligned_target, kPageSize, PROT_READ | PROT_WRITE | PROT_EXEC) != 0) {
+    // Log error on mprotect!
+    FLAMINGO_ABORT("Failed to mark: {} (page aligned: {}) as +rwx. err: {}", fmt::ptr(target),
+                   fmt::ptr(page_aligned_target), std::strerror(errno));
+  }
+  static flamingo::Trampoline target_hook(target, hookSize, trampoline_size);
+  // PlayerStartFirstScene has a bool parameter, my guess is that it is like the SceneManager::LoadFirstScene, where the bool param determines whether the load occurs as async
+  auto unity_hook = [](bool param) noexcept{
```

Comment:
Help request: At the moment this does not unhook.
 - Should it?
 - Does it matter? in theory PlayerStartFirstScene would only ever be called once is the assumption